src/main/java/com/sintern/service/ApplicationServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.sintern.service;
+
+import com.sintern.domain.entity.Application;
+import com.sintern.domain.entity.FileEntity;
+import com.sintern.domain.entity.OpenInternPosition;
+import com.sintern.domain.entity.Student;
+import com.sintern.exception.CVNotFoundException;
+import com.sintern.exception.EntityNotFoundException;
+import com.sintern.repository.ApplicationRepository;
+import com.sintern.repository.FileRepository;
+import com.sintern.repository.OpenInternPositionRepository;
+import com.sintern.repository.StudentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class ApplicationServiceImpl implements ApplicationService {
+
+    private final ApplicationRepository applicationRepository;
+    private final StudentRepository studentRepository;
+    private final OpenInternPositionRepository openInternPositionRepository;
+    private final FileRepository fileRepository;
+
+    @Autowired
+    public ApplicationServiceImpl(ApplicationRepository applicationRepository,
+                                  StudentRepository studentRepository,
+                                  OpenInternPositionRepository openInternPositionRepository,
+                                  FileRepository fileRepository) {
+        this.applicationRepository = applicationRepository;
+        this.studentRepository = studentRepository;
+        this.openInternPositionRepository = openInternPositionRepository;
+        this.fileRepository = fileRepository;
+    }
+
+    @Override
+    @Transactional
+    public void addApplication(UUID studentID, UUID openInternPositionID, String description) {
+        Student student = studentRepository.findStudentById(studentID);
+        if(student == null)
+            throw new EntityNotFoundException("Student with given id does not exist!");
+
+        Optional<FileEntity> fileEntity = fileRepository.findByStudentId(studentID);
+        if(fileEntity.isEmpty())
+            throw new CVNotFoundException("Student does not have attached a CV!");
+
+        OpenInternPosition openInternPosition = openInternPositionRepository.findOpenInternPositionById(openInternPositionID);
+        if(openInternPosition == null)
+            throw new EntityNotFoundException("Position with given id does not exist!");
+
+        Application applicationFound = applicationRepository.findApplicationByStudentIdAndOpenInternPositionId(studentID, openInternPositionID);
+        if(applicationFound != null){
+            applicationFound.setStudent(student);
+            applicationFound.setOpenInternPosition(openInternPosition);
+            applicationFound.setDateOfSubmission(LocalDateTime.now());
+            applicationFound.setDescription(description);
+
+            applicationRepository.save(applicationFound);
```

Comment:
If a student already applied to a position, we give him the possibility to apply again, changing both the date of submission and the description?

Reply:
I thought that if he applied with the wrong CV, he can reapply and just overwrite the data

Reply:
Yep, I think we can leave it this way, allowing the Student to apply again, by introducing other data (maybe he updated the CV a day later).